src/main/kotlin/com/ampnet/payoutservice/service/PayoutServiceImpl.kt

```diff
@@ -0,0 +1,70 @@
+package com.ampnet.payoutservice.service
+
+import com.ampnet.payoutservice.blockchain.BlockchainService
+import com.ampnet.payoutservice.controller.response.CreatePayoutResponse
+import com.ampnet.payoutservice.exception.ErrorCode
+import com.ampnet.payoutservice.exception.InvalidRequestException
+import com.ampnet.payoutservice.repository.MerkleTreeRepository
+import com.ampnet.payoutservice.util.BlockNumber
+import com.ampnet.payoutservice.util.ChainId
+import com.ampnet.payoutservice.util.ContractAddress
+import com.ampnet.payoutservice.util.HashFunction
+import com.ampnet.payoutservice.util.MerkleTree
+import com.ampnet.payoutservice.util.WalletAddress
+import mu.KLogging
+import org.springframework.stereotype.Service
+
+@Service
+class PayoutServiceImpl(
+    private val merkleTreeRepository: MerkleTreeRepository,
+    private val ipfsService: IpfsService,
+    private val blockchainService: BlockchainService
+) : PayoutService {
+
+    companion object : KLogging()
+
+    override fun createPayout(
+        chainId: ChainId,
+        assetAddress: ContractAddress,
+        requesterAddress: WalletAddress,
+        payoutBlock: BlockNumber
+    ): CreatePayoutResponse {
+        logger.info {
+            "Payout request for chain ID: $chainId, asset address: $assetAddress," +
+                " requester address: $requesterAddress, payout block: $payoutBlock"
+        }
+        val assetOwner = blockchainService.getAssetOwner(chainId, assetAddress)
+
+        if (assetOwner != requesterAddress) {
+            logger.warn { "Requester is not asset owner" }
+            throw InvalidRequestException(
+                ErrorCode.USER_NOT_ASSET_OWNER,
+                "User with wallet address: $requesterAddress is not the owner of asset contract: $assetAddress"
+            )
+        }
```

Comment:
I would wrap this check with some application property flag so we can disable it if we don't need it. We talk about importing existing erc20 tokens so this could cause us problems in the future.